ludwig/backend/ray.py

```diff
@@ -758,6 +760,19 @@ def check_lazy_load_supported(self, feature):
                 f"Set preprocessing config `in_memory: True` for feature {feature[NAME]}"
             )
 
+    def read_binary_files(self, column: Series, map_fn: Optional[Callable] = None) -> Series:
+        ds = self.df_engine.to_ray_dataset(column.to_frame(name=column.name))
+
+        def map_batches_fn(df: pd.DataFrame, fn: Callable) -> pd.DataFrame:
+            df[column.name] = df[column.name].map(fn)
+            return df
+
+        ds = ds.map_batches(partial(map_batches_fn, fn=get_bytes_obj_if_path), batch_format="pandas")
```

Comment:
This approach makes sense. One concern I had in the back of my mind is regarding partition size. For example, if the input dataset is like 1MB, and so it ends up in a single partition as a Dask DF, then it could explode in size if every image is 10MB being shoved into a single partition.

This is likely something we'll want to wait to see before we prematurely optimize it, but it's something we should be aware of.

Reply:
I see, that makes sense. We may want to repartition if that is the case, but we want to be careful in doing that to make sure that our processed columns can still be coerced into a dataframe again by the `df_like` call at the bottom of `data.preprocessing.build_dataset`: https://github.com/ludwig-ai/ludwig/blob/master/ludwig/data/preprocessing.py#L1137

Happy to follow-up on this later on if this becomes an issue.